packages/contracts-bedrock/go-fuzz/trie/trie.go

```diff
@@ -0,0 +1,193 @@
+package trie
+
+import (
+	"crypto/rand"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethdb/memorydb"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// Generate an abi-encoded `trieTestCase` of a specified variant
+func FuzzTrie(variant string) {
+	if len(variant) < 1 {
+		log.Fatal("Must pass a variant to the trie fuzzer!")
+	}
+
+	var testCase trieTestCase
+	switch variant {
+	case "valid":
```

Comment:
Swapped over to using enums- went with string variants so that they are readable within foundry tests w/o having to reference the variant docs.

i.e.

`ffi.getMerkleTrieFuzzCase("valid")` reads better than `ffi.getMerkleTrieFuzzCase(0)`